src/test/scala/com/microsoft/ml/spark/explainers/LeastSquaresRegressionSuite.scala

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.explainers
+
+import com.microsoft.ml.spark.core.test.base.TestBase
+import breeze.linalg.{DenseMatrix => BDM, DenseVector => BDV}
+import breeze.numerics.abs
+
+class LeastSquaresRegressionSuite extends TestBase {
+
+  test("LeastSquaresRegression should regress correctly with weights with intercept.") {
+    // Validated with the following code in Jupyter notebook:
+    // import numpy as np
+    // from sklearn.linear_model import LinearRegression
+    // x = np.array([[13, 4, 5], [8, 16, 7], [4, 24, 13], [24, 15, 27], [15, 9, 8]])
+    // y = np.array([2, 4, 3, 5, 1])
+    // weights = np.array([1, 2, 2, 1, 2])
+    // model = LinearRegression(fit_intercept = True).fit(x, y, sample_weight=weights)
+    // print(model.intercept_, model.coef_)
+
+    val x = BDM((13d, 4d, 5d), (8d, 16d, 7d), (4d, 24d, 13d), (24d, 15d, 27d), (15d, 9d, 8d))
+    val y = BDV[Double](2d, 4d, 3d, 5d, 1d)
+    val weights = BDV[Double](1d, 2d, 2d, 1d, 2d)
+
+    val result = new LeastSquaresRegression().fit(x, y, weights, fitIntercept = true)
+
+    assert(abs(result.intercept - 0.6050434048734146) < 1e-6)
+    assert(abs(result.coefficients(0) - 0.01518619) < 1e-6)
+    assert(abs(result.coefficients(1) - 0.08135246) < 1e-6)
+    assert(abs(result.coefficients(2) - 0.082494) < 1e-6)
+    assert(abs(result.rSquared - 0.43565371571037637) < 1e-6)
+    assert(abs(result.loss - 8.394650978808151) < 1e-6)
+  }
+
+  test("LeastSquaresRegression should regress correctly with weights without intercept.") {
+    // Validated with the following code in Jupyter notebook:
+    // import numpy as np
```

Comment:
not sure what you mean by this.

Reply:
You mean the comment or this test itself?
This test itself validates that the LeastSquaresRegression returns correct result when we specify non-trivial sample weights but without fitting intercept.

The comment here just documents the python code that runs the same regression with same inputs/outputs/weights, but runs the regression through sklearn. We expect the results to match between scala implementation and sklearn.